### ISSUE TITLE
Fine-tune

### ISSUE THREAD

**Issue by mkairanbay** [sklearn-theano/sklearn-theano#100]
I would like to fine-tune the Overfeat model with my own data where I have only 2 labels. In order to do this I change the number of nodes in last layer from 1000 to 2 in SMALL_NETWORK_FILTER_SHAPES and LARGE_NETWORK_FILTER_SHAPES. So, they look like these:
SMALL_NETWORK_FILTER_SHAPES = np.array([(96, 3, 11, 11),
                                        (256, 96, 5, 5),
                                        (512, 256, 3, 3),
                                        (1024, 512, 3, 3),
                                        (1024, 1024, 3, 3),
                                        (3072, 1024, 6, 6),
                                        (4096, 3072, 1, 1),
                                        (2, 4096, 1, 1)])
LARGE_NETWORK_FILTER_SHAPES = np.array([(96, 3, 7, 7),
                                        (256, 96, 7, 7),
                                        (512, 256, 3, 3),
                                        (512, 512, 3, 3),
                                        (1024, 512, 3, 3),
                                        (1024, 1024, 3, 3),
                                        (4096, 1024, 5, 5),
                                        (4096, 4096, 1, 1),
                                        (2, 4096, 1, 1)])

However, when I started to fit the network with my own data in the following way:
overfeat = OverfeatClassifier(top_n = 2)
overfeat.fit(X_train, Y_train)
I found the body of fit function is empty.

def fit(self, X, y=None):
        """Passthrough for scikit-learn pipeline compatibility."""
        return self

So, is there any other way to fine-tune the model? Thank you.

**Comment by jrosebr1** [comment]
The `sktheano-library` doesn't handle the actual *training* of a network. It takes pre-trained network architectures and their associated weights (such as OverFeat, VGG, GoogLeNet, etc.) and provides methods to load these networks and pass images through them for feature extraction, localization, etc.

If you would like to fine-tune a network, you'll need to utilize a library that handles the *training* of a network, such as Keras, Theano, Caffe, etc.

**Comment by kastnerkyle** [comment]
Fine tuning is a feature we have considered but have put a ways down the road. Fine tuning opens a "pandora's box" of advanced usage which may be difficult to support  without becoming Lasagne/Keras which doesn't seem overly useful to me at this time. We have a few ideas to try though before making any decision.
